src/flint/test/test_all.py

```diff
@@ -2995,65 +3038,89 @@ def quick_poly():
         assert raises(lambda: quick_poly().imul(P(ctx=ctx1)), IncompatibleContextError)
         assert raises(lambda: quick_poly().imul(None), NotImplementedError)
 
-        assert quick_poly() // mpoly({(1, 1): 1}) == mpoly({(1, 1): 4})
-        assert quick_poly() % mpoly({(1, 1): 1}) \
-            == mpoly({(1, 0): 3, (0, 1): 2, (0, 0): 1})
-        assert divmod(quick_poly(), mpoly({(1, 1): 1})) \
-            == (mpoly({(1, 1): 4}), mpoly({(1, 0): 3, (0, 1): 2, (0, 0): 1}))
-
-        assert 1 / P(1, ctx=ctx) == P(1, ctx=ctx)
-        assert quick_poly() / 1 == quick_poly()
-        assert quick_poly() // 1 == quick_poly()
-        assert quick_poly() % 1 == P(ctx=ctx)
-        assert divmod(quick_poly(), 1) == (quick_poly(), P(ctx=ctx))
+        if (P is flint.fmpz_mod_mpoly or P is flint.nmod_mpoly) and not ctx.is_prime():
+            assert raises(lambda: quick_poly() // mpoly({(1, 1): 1}), DomainError)
+            assert raises(lambda: quick_poly() % mpoly({(1, 1): 1}), DomainError)
+            assert raises(lambda: divmod(quick_poly(), mpoly({(1, 1): 1})), DomainError)
+        else:
+            assert quick_poly() // mpoly({(1, 1): 1}) == mpoly({(1, 1): 4})
+            assert quick_poly() % mpoly({(1, 1): 1}) \
+                == mpoly({(1, 0): 3, (0, 1): 2, (0, 0): 1})
+            assert divmod(quick_poly(), mpoly({(1, 1): 1})) \
+                == (mpoly({(1, 1): 4}), mpoly({(1, 0): 3, (0, 1): 2, (0, 0): 1}))
+
+        if (P is flint.fmpz_mod_mpoly or P is flint.nmod_mpoly) and not ctx.is_prime():
+            pass
+        else:
+            assert 1 / P(1, ctx=ctx) == P(1, ctx=ctx)
+            assert quick_poly() / 1 == quick_poly()
+            assert quick_poly() // 1 == quick_poly()
+            assert quick_poly() % 1 == P(ctx=ctx)
+            assert divmod(quick_poly(), 1) == (quick_poly(), P(ctx=ctx))
+
+            assert S(1) / P(1, ctx=ctx) == P(1, ctx=ctx)
+            assert quick_poly() / S(1) == quick_poly()
+            assert quick_poly() // S(1) == quick_poly()
+            assert quick_poly() % S(1) == P(ctx=ctx)
+            assert divmod(quick_poly(), S(1)) == (quick_poly(), P(ctx=ctx))
 
         if is_field:
-            assert quick_poly() / 3 == mpoly({(0, 0): S(1, 3), (0, 1): S(2, 3), (1, 0): S(1), (2, 2): S(4, 3)})
+            if (P is flint.fmpz_mod_mpoly or P is flint.nmod_mpoly):
+                assert quick_poly() / 3 == mpoly({(0, 0): S(34), (0, 1): S(68), (1, 0): S(1), (2, 2): S(35)})
+            else:
+                assert quick_poly() / 3 == mpoly({(0, 0): S(1, 3), (0, 1): S(2, 3), (1, 0): S(1), (2, 2): S(4, 3)})
```

Comment:
It should be possible to use e.g. `S(1)/3` here and not need to split the two cases.

Reply:
Yep it is thanks. This also revealed that I wasn't used `fmpz_mod` for the `fmpz_mod_mploy` tests, which has opened another rabbit whole of things to add.